variables.tf

```diff
@@ -169,6 +179,16 @@ variable "restrict_default_security_group" {
   description = "Set to true to remove all rules from the default security group"
 }
 
+variable "s3_interface_endpoint" {
+  type = object({
+    private_dns_enabled = bool
```

Comment:
```
Error: Error creating VPC Endpoint: InvalidParameter: Private DNS can't be enabled because the service com.amazonaws.yourregion.s3 does not provide a private DNS name.
        status code: 400, request id: whatever
```

it appears to be impossible to offer private dns for this service so we shouldn't offer it as a configurable.